PocketKit/Sources/PocketKit/AppBadgeTracker.swift

```diff
@@ -0,0 +1,70 @@
+import Foundation
+import Sync
+import UIKit
+import Combine
+
+protocol BadgeProvider: AnyObject {
+    var applicationIconBadgeNumber: Int { get set }
+}
+
+extension UIApplication: BadgeProvider { }
+
+class AppBadgeSetup {
+
+    private let source: Source
+    private let notificationCenter: NotificationCenter
+    private var subscriptions: Set<AnyCancellable> = []
+    private var userDefaults: UserDefaults
+    private let badgeProvider: BadgeProvider
+    private let completion: (() -> Void)?
```

Comment:
@timc-mozilla Can you leave a quick comment in code about why we added this? Here's an idea:

```
This completion block is called once the badge value is updated. This completion block is not currently used in the app code, but is utilized in tests, since setting the badge needs to occur on the main thread (when using UIApplication as the provider), which is called asynchronously. Thus, this is added as async test support.
```

Reply:
okee dokee